whynotearth/BrowTricks
# Feat: Add Default Image to LineItem Component

**Note: #47 must be merged before this PR**

# Description

This PR addresses #45 indirectly. We add a default for the image slot for the 'BaseLineItem' component.

We also make some other minor adjustments:

* Add typography classes to elements that wrap the slots, as these will not vary from instantiation to instantiation. Doing this will reduce the code in 'BaseLineItem' component instantiations.

* Wrap the 'detail' portion in an `a` tag for accessibility, as it has a click event listener on it

# Commits

5801bc8: Add the default line-item icon asset

0664a20: Implement the default line-item icon in 'BaseLineItem' component

45d8aaf: Wrap the clickable 'detail' part of the 'BaseLineItem' component in an `a` tag for accessibility

4d3b77a: Add typography classes to some slots in 'BaseLineItem' component to help brevity in instantiations

c1751a7: Use more explanatory import names (e.g. IconLineItemDefault instead of LineItemImageDefault)

ba22699: Example implementation of 'BaseLineItem' component using new default icon

# How To Test

This PR does not introduce visual changes, only functional. I included a 'BaseLineItem' component instantiation in the home view of customers.

Visit the customer home page ([link to Netlify deploy](https://deploy-preview-49--browtricks.netlify.app/)) and notice that the 'BaseLineItem' component still looks the same as before, now just with default icon, matching the Figma designs:

![image](https://user-images.githubusercontent.com/20192983/81635899-48f4d000-93c7-11ea-9b15-cfd6cd6e53dd.png)
